src/period-calculation/generate-fixed-periods/generate-fixed-periods-weekly.ts

```diff
@@ -22,29 +20,34 @@ const getStartingDay = (
     startingDay: number | undefined
 ) => {
     switch (periodType) {
-        case 'WEEKLY':
-        case 'BIWEEKLY':
+        case FIXED_PERIOD_TYPES.WEEKLY:
+        case FIXED_PERIOD_TYPES.BIWEEKLY:
             return startingDay || Days.Monday
-        case 'WEEKLYSAT':
+        case FIXED_PERIOD_TYPES.WEEKLYSAT:
             return Days.Saturday
-        case 'WEEKLYSUN':
+        case FIXED_PERIOD_TYPES.WEEKLYSUN:
             return Days.Sunday
-        case 'WEEKLYTHU':
+        case FIXED_PERIOD_TYPES.WEEKLYTHU:
             return Days.Thursday
-        case 'WEEKLYWED':
+        case FIXED_PERIOD_TYPES.WEEKLYWED:
             return Days.Wednesday
         default:
             throw new Error(`unrecoginsed weekly period type: ${periodType}`)
     }
 }
 
-export const getWeeklyPeriods: GeneratedPeriodsFunc = ({
+// Does not need a `locale` as we're displaying the month as number in the
+// displayName/name
+const generateFixedPeriodsWeekly: GeneratedPeriodsFunc = ({
     year,
     calendar,
     periodType,
     startingDay,
-    locale = 'en-GB',
+    excludeDay: _excludeDay,
```

Comment:
Sorry, I just realised that this isn't right. It's not a breaking change because you are adding an _optional_ parameter. It'd be a feature bump.

Reply:
This can be used to generate periods up to a certain point.
Let's say you want to display a list of all periods of this year which are not open (current date > end date), e.g. what the aggregate data entry app does. Now you can simply provide the current date.

Do you want me to add that as a comment to the type definition?